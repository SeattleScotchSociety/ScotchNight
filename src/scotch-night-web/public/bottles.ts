import { IBottle } from "../src/app/stores/BottleStore";

export const bottles: IBottle[] = [
    {
        id: "bbf70a5b-4257-498f-9afe-7dac23d40496",
        distillery: "Westland",
        name: "Peated",
        age: 10,
        description: "I bought this because I like smoke."
    },
    {
        id: "dd979a80-293a-40a2-a936-3f5adfd0e92b",
        distillery: "Laphroaig",
        name: "18 Year",
        age: 18,
        description: ""
    }
];
export default bottles;
