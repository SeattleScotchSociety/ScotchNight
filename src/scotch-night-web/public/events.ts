import { IEvent } from "../src/app/stores/EventStore";

export const events: IEvent[] = [
    {
        id: "8ff33e2f-642c-4196-8c8e-c2e2b60a1205",
        date: "5/30/2017",
        title: "The First",
        location: "a15a7cb9-6cf7-4e97-900c-32b3f0427263",
        public: false,
        description: "The very first Seattle Scotch Society night. Join us for scotch and pizza!"
    },
    {
        id: "93aec433-52f1-4cbf-824e-c19397359109",
        title: "The Second",
        date: "8/5/2017",
        location: "5b14026b-0da2-492c-88ee-68df1aebcdb1",
        public: false,
        description: `The first was a success, now let's keep them coming... onto our second scotch night!`,
        menu: [
            "bbf70a5b-4257-498f-9afe-7dac23d40496",
            "dd979a80-293a-40a2-a936-3f5adfd0e92b"
        ]
    }
];
export default events;
