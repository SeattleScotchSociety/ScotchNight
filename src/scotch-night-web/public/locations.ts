import { ILocation } from "../src/app/stores/LocationStore";

export const locations: ILocation[] = [
    {
        id: "a15a7cb9-6cf7-4e97-900c-32b3f0427263",
        name: "The Phillabaum's",
        address1: "438 3rdth Ave W",
        address2: "",
        city: "Seattle",
        state: "WA",
        zipCode: "94299",
        position: {
            lat: 51.545955,
            lng: -0.1020112
        }
    },
    {
        id: "5b14026b-0da2-492c-88ee-68df1aebcdb1",
        name: "The Fix's",
        address1: "6780 70rth Ave NW",
        address2: "",
        city: "Seattle",
        state: "WA",
        zipCode: "98113",
        position: {
            lat: 51.545955,
            lng: -0.1020112
        }
    },
    {
        id: "bb59bcf4-6658-4c0c-bf5b-c373451c78a5",
        name: "The Merkel's",
        address1: "128 NE 83stth St.",
        address2: "",
        city: "Kirkland",
        state: "WA",
        zipCode: "98323",
        position: {
            lat: 51.545955,
            lng: -0.1020112
        }
    }
];