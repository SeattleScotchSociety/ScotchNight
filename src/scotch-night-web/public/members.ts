import { IMember } from "../src/app/stores/MemberStore";

export const members: IMember[] = [
    {
        id: "131bbea4-15ba-42ac-a02d-983a9b612fc1",
        firstName: "Shannon",
        lastName: "Andrews",
        email: "andrews.shannon@gmail.com"
    },
    {
        id: "4624f9d3-56cb-426c-a0aa-2a2541bd9e7f",
        firstName: "Travis",
        lastName: "Merkel",
        email: "tmerkel@live.com"
    },
    {
        id: "2d82f2d0-8340-45bf-ab6b-394fd056a0c8",
        firstName: "Nicole",
        lastName: "Cua",
        email: "ncua@mac.com"
    },
    {
        id: "a8ea6c21-3e01-46cb-ace2-3ab2f2d71df2",
        firstName: "Adam",
        lastName: "Phillabaum",
        email: "adam@phillabaum.us"
    },
    {
        id: "048cee6f-31c4-45ec-a0e5-b76c3056d51b",
        firstName: "Louis",
        lastName: "Han",
        email: "louishan83@gmail.com"
    }
];
