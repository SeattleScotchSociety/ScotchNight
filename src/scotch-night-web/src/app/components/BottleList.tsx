import { inject, observer } from "mobx-react";
import * as React from "react";
import * as _ from 'lodash';

import { IRootStore } from "../stores/RootStore";

interface IBottleListProps { store: IRootStore; }

@inject("store")
@observer
export class BottleList extends React.Component<IBottleListProps> {
    constructor(props: IBottleListProps) {
        super(props);

        this._handleOnClick = this._handleOnClick.bind(this);
    }

    private _handleOnClick(bottle) {
        const { navigation, scotchNightStore } = this.props.store;

        scotchNightStore.setCurrentBottle(bottle);
        navigation.push(`/bottles/${bottle.id}`);
    }

    public render() {
        const { bottles } = this.props.store.bottleStore;
        const bottleElements = _.sortBy(bottles, ['distillery', 'name']).map((bottle) => {
            const { name, distillery } = bottle;

            return (
                <div className="card" key={bottle.id.toString()} onClick={() => this._handleOnClick(bottle)}>
                    <h3 className="card__head">{distillery}</h3>
                    <h4 className="card__subhead">{name}</h4>
                </div>
            );
        });

        return (
            <div className="container">
                <h1 className="margin-bottom-md">Scotch Library</h1>
                {bottleElements}
            </div>
        );
    }
}

export default BottleList;
