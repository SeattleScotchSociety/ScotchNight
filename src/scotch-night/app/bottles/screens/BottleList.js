// @flow
import React from 'react';
import { StyleSheet, Button, View, FlatList, ActivityIndicator } from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { loadBottles } from '../BottleActions';
import BottleListItem from '../components/BottleListItem';

class BottleList extends React.Component {
    static navigationOptions = {
        title: 'Bottles'
    };

    componentWillMount() {
        let { loadBottles } = this.props.actions;

        loadBottles();
    }

    renderHeader = () => {
        return (
            <View style={{ flex: 1 }}>
                <Button title="Add Bottle" onPress={this._onAddPress.bind(this)} />
            </View>
        );
    };

    renderFooter = () => {
        if (!this.props.loading) return null;

        return (
            <View
                style={{
                    paddingVertical: 20,
                    borderTopWidth: 1,
                    borderColor: '#CED0CE'
                }}>
                <ActivityIndicator animating size="large" />
            </View>
        );
    };

    _onAddPress() {
        const { navigate } = this.props.navigation;

        navigate('AddBottle', {});
    }

    render() {
        const { navigate } = this.props.navigation;

        return (
            <View style={{ flex: 1 }}>
                <FlatList
                    data={this.props.bottles || []}
                    renderItem={({ item }) => (
                        <BottleListItem
                            bottle={{ ...item }}
                            onPress={() => navigate('BottleDetail', { bottle: { ...item } })}
                        />
                    )}
                    keyExtractor={item => item.id}
                    ListHeaderComponent={this.renderHeader}
                    ListFooterComponent={this.renderFooter}
                />
            </View>
        );
    }
}

function mapStateToProps(state) {
    let { bottles } = state;

    return { bottles };
}

function mapDispatchToProps(dispatch) {
    return { actions: bindActionCreators({ loadBottles }, dispatch) };
}

export default connect(mapStateToProps, mapDispatchToProps)(BottleList);

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center'
    }
});
