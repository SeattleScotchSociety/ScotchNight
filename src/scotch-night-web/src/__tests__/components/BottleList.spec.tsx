import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import React from "react";
import ReactDOM from "react-dom";

import { locations } from "../../../public/locations";
import BottleList from "../../app/components/BottleList";

import { rootStore } from "../mocks";

configure({ adapter: new Adapter() });

describe("Component: BottleList", () => {
    it("should render without throwing an error", () => {
        const wrapper = shallow(
            <BottleList
                store={rootStore}
            />
        ).toJson;

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});