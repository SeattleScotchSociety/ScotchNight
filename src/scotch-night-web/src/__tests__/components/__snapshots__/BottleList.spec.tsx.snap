// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: BottleList should render without throwing an error 1`] = `undefined`;
